sql/log_event.cc

```diff
@@ -9322,24 +9384,25 @@ int Rows_log_event::do_apply_event(Relay_log_info const *rli) {
       if (thd->slave_thread) {
         rli->report(ERROR_LEVEL, error, ER_THD(thd, error), buf);
         thd->is_slave_error = 1;
-        const_cast<Relay_log_info *>(rli)->slave_close_thread_tables(thd);
       } else {
         /*
           For the cases in which a 'BINLOG' statement is set to
           execute in a user session
         */
         my_printf_error(error, ER_THD(thd, error), MYF(0), buf);
       }
-      DBUG_RETURN(error);
+      /* remove trigger's tables */
+      goto err;
     }
   }
 
   table = m_table =
       const_cast<Relay_log_info *>(rli)->m_table_map.get_table(m_table_id);
 
-  DBUG_PRINT("debug",
-             ("m_table: %p, m_table_id: %llu", m_table, m_table_id.id()));
-
+  DBUG_PRINT(
+      "debug",
+      ("m_table: 0x%lx, m_table_id: %llu%s", (ulong)m_table, m_table_id.id(),
```

Comment:
Printing pointer as pointer and not as 0x%lx was good already

```suggestion
      ("m_table: %p, m_table_id: %llu%s", m_table, m_table_id.id(),
```